R/clean_spelling.R

```diff
@@ -0,0 +1,96 @@
+#' Rename values in a vector based on a wordlist
+#'
+#' This function provides an interface for [forcats::fct_recode()], 
+#' [forcats::fct_explicit_na()], and [forcats::fct_relevel()] in such a way that
+#' a data wordlist can be imported from a data frame. 
+#'
+#' @param x a character or factor vector
+#' @param wordlist a two-column matrix or data frame defining mis-spelled
+#'   words in the first column and replacements in the second column.
+#' 
+#' @return a vector of the same type as `x` with mis-spelled labels cleaned. 
+#'   Note that factors will be arranged by the order presented in the data 
+#'   wordlist; other levels will appear afterwards.  
+#'
+#' @author Zhian N. Kamvar
+#' @seealso [clean_variable_spelling()] for an implementation that acts across
+#'   multiple variables in a data frame.
+#' @export
+#' @examples
+#'
+#' corrections <- data.frame(
+#'   bad = c("foubar", "foobr", "fubar", NA, "unknown"),
+#'   good = c("foobar", "foobar", "foobar", "missing", "missing"),
+#'   stringsAsFactors = FALSE
+#' )
+#' head(corrections)
+#' my_data <- c(letters[1:5], sample(corrections$bad, 10, replace = TRUE))
+#' clean_spelling(my_data, corrections)
+#'
+#' # The function will give you a warning if the wordlist does not
+#' # match the data
+#' clean_spelling(letters, corrections)
+#'
+#' # The can be used for translating survey output
+#'
+#' dict <- data.frame(
+#'   option_code = c("Y", "N", "U", NA),
+#'   option_name = c("Yes", "No", "Unknown", "Missing"),
+#'   stringsAsFactors = FALSE
+#' )
+#' clean_spelling(c("Y", "Y", NA, "N", "U", "U", "N"), dict)
+#'
+#' @importFrom forcats fct_recode fct_explicit_na fct_relevel
+#' @importFrom rlang "!!!"
+
+clean_spelling <- function(x = character(), wordlist = data.frame()) {
+
+  if (length(x) == 0 || !is.atomic(x)) {
+    stop("x must be coerceable to a character")
+  } else if (!is.factor(x)) {
+    x <- as.character(x)
+  }
+
+  if (length(wordlist) < 2 || !is.data.frame(wordlist)) {
```

Comment:
Will this work if the wordlist is provided as a matrix?

Reply:
No, but I can fix that :)